### PR TITLE
Network file dropins

### PR DESCRIPTION
**Pull request by keszybz**
This cleans up a bit the implementation in #3655 and extends it to .netdev files.

### CODE REVIEW

**Comment by keszybz** [comment]
Actually #3716.

**Comment by martinpitt** [comment]
This breaks the networkd tests that use networkd to bring up the "router" end. Bringing it up with dnsmasq seems ok. This looks real and should be investigated first before merging, so setting the "fail" tag for now.

File: test/networkd-test.py [hunk 1]
```diff
@@ -461,6 +462,40 @@ def test_search_domains_too_long(self):
...
+Name=dummy0
+Kind=dummy
+MACAddress=12:34:56:78:9a:bc''')
+        with open('/run/systemd/network/test.network', 'w') as f:
+            f.write('''[Match]
+Name=dummy0
+[Network]
+Address=192.168.42.100
+DNS=192.168.42.1''')
+        os.mkdirs('/etc/systemd/network/test.network.d', exist_ok=True)
```
Comment (martinpitt) [inline test/networkd-test.py]: It's `os.makedirs()`. This will at least fix the `AttributeError`.

**Comment by martinpitt** [comment]
Works fine now, thanks for fixing! Dropping the CI-failure label now. (s390x is still stalled, looking into this)

File: test/networkd-test.py [hunk 2]
```diff
@@ -323,7 +323,6 @@ def create_iface(self, ipv6=False):
             f.write('''#!/bin/sh -eu
 mkdir -p /run/systemd/network
 mkdir -p /run/systemd/netif
-mkdir -p /run/systemd/network/test.network.d
```
Comment (martinpitt) [inline test/networkd-test.py]: This line just got added in the previous commit https://github.com/systemd/systemd/pull/4123/commits/d831ec1f, so they cancel each other out.

**Comment by keszybz** [comment]
Pushed with the suggested changes.

**Comment by Sufflope** [comment]
Hello,

sorry for letting the initial PR go stale ; I just didn't manage to run the failing tests:
- tried in a nspawn container but then couldn't run tests because of missing loopback devices
- tried directly on my F24 server without isolation in a container, but TEST-01-BASIC was stuck booting the VM (I waited for hours before ctrl-c)
- tried on a qemu Ubuntu 16.04 VM to be closer to the integration environment but had the same results as on my server (test VM stuck at boot)

I guess I'll never know why I just can't manage to run them.

Glad though the feature made it in master, thanks a lot!

**Pull request merged**